From 60a6eece06bde26516649bdcbed4096dd734503e Mon Sep 17 00:00:00 2001
From: Todd Lipcon <todd@cloudera.com>
Date: Mon, 4 Oct 2010 18:01:39 -0700
Subject: [PATCH 0726/1179] CLOUDERA-BUILD. Send SecurityAudit logs to the console unless running through hadoop-daemon.sh

Reason: Fixes issue where clients would try to write SecurityAuth.audit logs
Author: Todd Lipcon
---
 bin/hadoop-daemon.sh  |    1 +
 conf/log4j.properties |    3 ++-
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/bin/hadoop-daemon.sh b/bin/hadoop-daemon.sh
index 5ad930f..d502099 100755
--- a/bin/hadoop-daemon.sh
+++ b/bin/hadoop-daemon.sh
@@ -85,6 +85,7 @@ fi
 # some variables
 export HADOOP_LOGFILE=hadoop-$HADOOP_IDENT_STRING-$command-$HOSTNAME.log
 export HADOOP_ROOT_LOGGER="INFO,DRFA"
+export HADOOP_SECURITY_LOGGER="INFO,DRFAS"
 export _HADOOP_DAEMON_OUT=$HADOOP_LOG_DIR/hadoop-$HADOOP_IDENT_STRING-$command-$HOSTNAME.out
 export _HADOOP_DAEMON_PIDFILE=$HADOOP_PID_DIR/hadoop-$HADOOP_IDENT_STRING-$command.pid
 export _HADOOP_DAEMON_DETACHED="true"
diff --git a/conf/log4j.properties b/conf/log4j.properties
index 54576d9..cf97232 100644
--- a/conf/log4j.properties
+++ b/conf/log4j.properties
@@ -1,5 +1,6 @@
 # Define some default values that can be overridden by system properties
 hadoop.root.logger=INFO,console
+hadoop.security.logger=INFO,console
 hadoop.log.dir=.
 hadoop.log.file=hadoop.log
 
@@ -79,7 +80,7 @@ log4j.appender.DRFAS.File=${hadoop.log.dir}/${hadoop.security.log.file}
 log4j.appender.DRFAS.layout=org.apache.log4j.PatternLayout
 log4j.appender.DRFAS.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
 #new logger
-log4j.category.SecurityLogger=INFO,DRFAS
+log4j.category.SecurityLogger=${hadoop.security.logger}
 
 #
 # Rolling File Appender
-- 
1.7.0.4

